src/main/java/com/spotify/google/cloud/pubsub/client/Publisher.java

```diff
@@ -229,22 +257,57 @@ private TopicQueue(final String topic) {
         }
       } while (!size.compareAndSet(currentSize, newSize));
 
+      // Enqueue outgoing message
       queue.add(new QueuedMessage(message, future));
 
-      send();
+      // Schedule future batch sending
+      scheduleSend(newSize);
 
       return future;
     }
 
     /**
-     * Attempt to start sending this topic. If the concurrency level of the publisher has already been reached, enqueue
-     * this topic onto the pending queue for later sending.
+     * Schedule this topic for future enqueuing for batch sending. If the batch size has been reached, enqueue for
+     * sending immediately.
+     *
+     * @param queueSize The current number of enqueued messages in this topic.
      */
-    private void send() {
+    private void scheduleSend(final int queueSize) {
+
+      // Bail if this topic is already enqueued for sending.
       if (pending) {
         return;
       }
 
+      // Reached the batch size? Enqueue topic for sending immediately.
+      if (queueSize >= batchSize) {
+        enqueueSend();
+        return;
+      }
+
+      // Already scheduled this topic? Bail.
+      if (scheduled) {
+        return;
+      }
+
+      // Schedule this topic for later enqueuing, allowing more messages to gather into a larger batch.
+      try {
+        scheduled = true;
+        scheduler.schedule(this::enqueueSend, maxLatencyMs, MILLISECONDS);
```

Comment:
Redundantly scheduling multiple times is fine, but I'll make this atomic to avoid it.